include/mbedtls/ssl_ciphersuites.h

```diff
@@ -373,10 +373,8 @@ struct mbedtls_ssl_ciphersuite_t
     mbedtls_md_type_t MBEDTLS_PRIVATE(mac);
     mbedtls_key_exchange_type_t MBEDTLS_PRIVATE(key_exchange);
 
-    int MBEDTLS_PRIVATE(min_major_ver);
-    int MBEDTLS_PRIVATE(min_minor_ver);
-    int MBEDTLS_PRIVATE(max_major_ver);
-    int MBEDTLS_PRIVATE(max_minor_ver);
+    uint16_t MBEDTLS_PRIVATE(min_tls_version);
```

Comment:
This is a private member.  Using uint16_t (2 bytes) takes up less space than the int-sized enum (4 bytes) in `mbedtls_ssl_ciphersuite_t`, which is used in `library/ssl_ciphersuites.c:ciphersuite_definitions[]`, a static array of 191 elements.  Please confirm you would like to use the extra space, or if you would prefer that I use a bitfield, e.g. `bedtls_ssl_protocol_version MBEDTLS_PRIVATE(min_tls_version) : 16;`  (bitfields could conceivably be used for other members in `mbedtls_ssl_ciphersuite_t`, too)

Reply:
You seem to be quite sure that the enum is int-sized. The C spec allows compilers to use a smaller size if all the values in the enum fit in a smaller number of bytes: 2 bytes here for the `mbedtls_ssl_protocol_version` enum type. Is it the case that from your experience compilers do not do this ?

Reply:
In my experience across gcc, clang, Solaris cc, AIX xcc, and HPUX acc, an enum is int-sized unless explicitly specified with a compiler flag since doing otherwise may break binary compatibility, e.g. between a library and an application.  The gcc manual explains:
```
       -fshort-enums
           Allocate to an "enum" type only as many bytes as it needs for the declared range of possible values.  Specifically, the "enum" type is equivalent to the smallest integer type that
           has enough room.

           Warning: the -fshort-enums switch causes GCC to generate code that is not binary compatible with code generated without that switch.  Use it to conform to a non-default application
           binary interface.
```

Reply:
:rocket: implemented as (signed bitfield) `mbedtls_ssl_protocol_version MBEDTLS_PRIVATE(min_tls_version) : 16;`